tools/HiveToScripts/Program.cs

```diff
@@ -0,0 +1,170 @@
+using System.Collections.Concurrent;
+using System.Text.RegularExpressions;
+
+bool doLog = true;
+
+if (args.Intersect(["--help", "-h", "help", "?"]).Any())
+{
+    Console.WriteLine(
+@"Transforms hive test execution output into sh/ps1/http scripts with big inputs dumped as separate files:
+
+    ./HiveToScripts.exe [<path-to-hive-logs> [<output-path> [<json-rpc-url>]]]
+
+Example:
+
+    ./HiveToScripts.exe T:\logs T:\output http://localhost:8551
+");
+    return;
+}
+
+string hiveLogsDirectory = args.Length > 0 ? args[0] : "logs";
+
+if (Directory.Exists(Path.Combine(hiveLogsDirectory, "logs")))
+{
+    hiveLogsDirectory = Path.Combine(hiveLogsDirectory, "logs");
+}
+
+string outputDirectory = args.Length > 1 ? args[1] : "output";
+string jsonrpcUrl = args.Length > 2 ? args[2] : "http://localhost:8551";
+
+string[] simulatorFilePaths = Directory.GetFiles(hiveLogsDirectory, "*-simulator-*.log").Select(f => Path.Combine(outputDirectory, f)).ToArray();
+ConcurrentDictionary<string, TextWriter> fileDescriptors = new();
+
+foreach (string simulatorFilePath in simulatorFilePaths)
+{
+    using StreamReader simulatorFile = File.OpenText(simulatorFilePath);
+
+    if (Directory.Exists(outputDirectory))
+    {
+        foreach (string f in Directory.EnumerateFiles(outputDirectory))
+        {
+            File.Delete(f);
```

Comment:
This will attempt to delete files which you still have handles to in `fileDescriptors `

Reply:
Yeah, should be done outside of the foreach